File: 009__Palindrome_Number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        if x == 0:
            return True
        stack = []
        while x != 0:
            stack = [x % 10] + stack
            x /= 10
        for i in range(len(stack) / 2):
            if stack[i] != stack[-i-1]:
                return False
        return True
        

File: Java/009__Palindrome_Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int reverse = 0, origin = x;
        while (origin != 0) {
            reverse = reverse * 10 + origin % 10;
            origin /= 10;
        }
        return reverse == x;
    }
}
